### ISSUE TITLE
Should be artifactId "workspace" instead of one-jar-boot

### ISSUE THREAD

**Issue by skibu** [Kevin-Lee/one-jar-boot#3]
I'm trying out your solution and followed the documentation on how to modify pom file for maven. But ran into an error. I hunted around at http://dl.bintray.com/kevinlee/maven and it appears that the artifactId should be "workspace" instead of one-jar-boot.  Is that correct? If so it appears the documentation on your github page should be updated.

**Comment by kevin-lee** [comment]
@skibu Hmmmm it should be `one-jar-boot`. OK, I just checked the repo and found it's `workspace` as you said. I'm working now so will fix it later today. Thank you for reporting. :smile: :+1:

**Comment by kevin-lee** [comment]
By fixing, I mean changing `workspace` to `one-jar-boot` as `workspace` doesn't make any sense for the artifact id.

**Comment by kevin-lee** [comment]
@skibu OK, done! Please check out the README. Notice that the version is different now.

**Comment by skibu** [comment]
Yes, I can download and compile via maven when using the new artifactId and version. Thanks for fixing this.

**Comment by kevin-lee** [comment]
@skibu Not at all. Thank you for reporting and testing. :)

**Comment by skibu** [comment]
I was able to get the system to compile via maven using the new artifactId and version, but I still get the same problem using Amazon SDK via an application compiled using OneJar. I looked in my executable jar file and see that lib/one-jar-boot-0.97.3.jar is listed right at the top. But when accessing AWS SDK I still get the error "Caused by: java.lang.IllegalStateException: Fatal: Failed to load the internal config for AWS Java SDK".

Can you send me a sample Maven pom file that creates a OneJar jar file and that can successfully use AWS SDK?

**Comment by kevin-lee** [comment]
@skibu I fixed the issue because I wanted to use https://github.com/Kevin-Lee/sbt-onejar which uses `one-jar-boot` internally.

So I don't have any maven pom file that uses this `one-jar-boot` library.
However, if you can provide your `pom.xml` file or how you use it, I'll try to figure it out. Could you do that? Also, can you tell me if you use `onejar-maven-plugin`?

**Comment by skibu** [comment]
I'm indeed using onejar-maven-plugin, not sbt-onejar. Perhaps that is why I couldn't get one-jar-boot to work. I don't understand the purpose of sbt-onejar compared to onjar-maven-plugin. What is your motivation for using sbt-onejar?

The maven pop.xml file I am using is the following:

---

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
        <modelVersion>4.0.0</modelVersion>
        <groupId>transitime</groupId>
        <artifactId>transitimeCore</artifactId>
        <version>0.0.1-SNAPSHOT</version>

```
    <properties>
  <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
</properties>

    <repositories>
            <!--  Needed to fix problem with one-jar not working with AWS SDK.
                  Was getting error message "Fatal: Failed to load the internal
                  config for AWS"-->
            <repository>
            <snapshots>
                   <enabled>false</enabled>
            </snapshots>
            <id>bintray-kevinlee-maven</id>
            <name>bintray</name>
            <url>http://dl.bintray.com/kevinlee/maven</url>
            </repository>
    </repositories>
```

  <pluginRepositories>
                <pluginRepository>
                        <id>onejar-maven-plugin.googlecode.com</id>
                        <url>http://onejar-maven-plugin.googlecode.com/svn/mavenrepo</url>
                </pluginRepository>
        </pluginRepositories>
        <dependencies>
                
                <dependency>
                <groupId>com.simontuffs</groupId>
                <artifactId>one-jar-boot</artifactId>
                <version>0.97.3</version>
                </dependency>

```
            <!-- Database related, including c3p0 production quality connector.
                 Note: have found that can have problems if don't use similar
                 versions. Can have trouble finding the c3p0 connection pooler
                 for example if use old hibernate-core but new c3p0. -->
            <dependency>
                    <groupId>mysql</groupId>
                    <artifactId>mysql-connector-java</artifactId>
                    <version>5.1.35</version>
            </dependency>
            <dependency>
                    <groupId>org.postgresql</groupId>
                    <artifactId>postgresql</artifactId>
                    <version>9.3-1103-jdbc41</version>
            </dependency>
            <dependency>
                    <groupId>org.hibernate</groupId>
                    <artifactId>hibernate-core</artifactId>
                    <version>4.3.9.Final</version>
            </dependency>
            <dependency>
                    <groupId>org.hibernate.common</groupId>
                    <artifactId>hibernate-commons-annotations</artifactId>
                    <version>4.0.5.Final</version>
            </dependency>
            <dependency>
                    <groupId>org.hibernate</groupId>
                    <artifactId>hibernate-c3p0</artifactId>
                    <version>4.3.9.Final</version>
            </dependency>

            <!-- hibernate-core loads in really old version of xml-apis so
                 load in newer version explicitly. Otherwise get problems
                 with the Node class. -->
            <dependency>
                    <groupId>xml-apis</groupId>
                    <artifactId>xml-apis</artifactId>
                    <version>1.4.01</version>
            </dependency>

            <!-- For logging -->
            <dependency>
                    <groupId>ch.qos.logback</groupId>
                    <artifactId>logback-core</artifactId>
                    <version>1.1.2</version>
            </dependency>
            <dependency>
                    <groupId>ch.qos.logback</groupId>
                    <artifactId>logback-classic</artifactId>
                    <version>1.1.2</version>
            </dependency>
            <dependency>
                    <groupId>org.slf4j</groupId>
                    <artifactId>slf4j-api</artifactId>
                    <version>1.7.2</version>
            </dependency>

            <!-- So can use Amazon API for archiving files and such -->
            <dependency>
                    <groupId>com.amazonaws</groupId>
                    <artifactId>aws-java-sdk</artifactId>
                    <version>1.9.13</version>
            </dependency>

            <!-- For many java utilities, such as StringEscapeUtils -->
            <dependency>
                    <groupId>org.apache.commons</groupId>
                    <artifactId>commons-lang3</artifactId>
                    <version>3.3.2</version>
            </dependency>

            <!-- For processing XML files, such as for AVL feeds -->
            <dependency>
                    <groupId>org.jdom</groupId>
                    <artifactId>jdom</artifactId>
                    <version>2.0.2</version>
            </dependency>

            <!-- For GTFS-realtime feed -->
            <dependency>
                    <groupId>com.google.transit</groupId>
                    <artifactId>gtfs-realtime-bindings</artifactId>
                    <version>0.0.4</version>
            </dependency>

            <!-- For more easily handling command line options -->
            <dependency>
                    <groupId>commons-cli</groupId>
                    <artifactId>commons-cli</artifactId>
                    <version>1.2</version>
            </dependency>

            <!-- I believe not needed
            <dependency>
                    <groupId>org.apache.lucene</groupId>
                    <artifactId>lucene-core</artifactId>
                    <version>3.6.2</version>
            </dependency>
            -->

            <!-- For JMS for fancy handling of AVL feeds -->
            <dependency>
                    <groupId>org.hornetq</groupId>
                    <artifactId>hornetq-core-client</artifactId>
                    <version>2.3.25.Final</version>
            </dependency>
            <dependency>
                    <groupId>org.hornetq</groupId>
                    <artifactId>hornetq-jms-client</artifactId>
                    <version>2.3.25.Final</version>
            </dependency>

            <!-- For Amigocloud web sockets AVL feed  -->
            <dependency>
            <groupId>org.java-websocket</groupId>
                    <artifactId>Java-WebSocket</artifactId>
            <version>1.3.0</version>
            </dependency>

            <!--  For processing JSON. Needed for Amigocloud and other JSON based AVL feeds.
                  NOTE: the version 20141113 does not work with Java 1.7, which is what is
                  still the normal version of Java for AWS, at least as of June 2015. Need
                  to use old version 20140107 which was compiled with Java 1.7 instead
                  of Java 1.8 -->
            <dependency>
                <groupId>org.json</groupId>
            <artifactId>json</artifactId>
            <version>20140107</version>
            </dependency>

            <!-- For parsing CSV files, such as GTFS ones -->
            <dependency>
                    <groupId>org.apache.commons</groupId>
                    <artifactId>commons-csv</artifactId>
                    <version>1.1</version>
            </dependency>

            <!-- For concurrency annotations like @Immutable and @ThreadSave
                 from the "Concurrency in Practice" book need library -->
            <dependency>
                    <groupId>net.jcip</groupId>
                    <artifactId>jcip-annotations</artifactId>
                    <version>1.0</version>
            </dependency>

            <!-- For encryption/decryption -->
            <dependency>
                    <groupId>org.jasypt</groupId>
                    <artifactId>jasypt</artifactId>
                    <version>1.9.2</version>
            </dependency>

            <!-- So can send out monitoring related e-mails -->
            <dependency>
                    <groupId>javax.mail</groupId>
                    <artifactId>mail</artifactId>
                    <version>1.4.7</version>
            </dependency>

            <!-- JUnit testing of course -->
            <dependency>
                    <groupId>junit</groupId>
                    <artifactId>junit</artifactId>
                    <version>4.11</version>
                    <scope>test</scope>
            </dependency>

            <!-- Servlet related functionality -->
            <dependency>
                    <groupId>javax.servlet</groupId>
                    <artifactId>javax.servlet-api</artifactId>
                    <version>3.0.1</version>
                    <scope>provided</scope>
            </dependency>
    </dependencies>

    <build>
            <plugins>
                    <plugin>
                            <artifactId>maven-compiler-plugin</artifactId>
                            <version>2.5.1</version>
                            <configuration>
                                    <source>1.7</source>
                                    <target>1.7</target>
                            </configuration>
                    </plugin>
                    <plugin>
                            <groupId>org.dstovall</groupId>
                            <artifactId>onejar-maven-plugin</artifactId>
                            <version>1.4.4</version>
                            <executions>
                                    <execution>
                                            <id>build-first</id>
                                            <configuration>
                                                    <mainClass>org.transitime.applications.Core</mainClass>
                                                    <attachToBuild>true</attachToBuild>
                                                    <classifier>onejar</classifier>
                                                    <filename>core.jar</filename>
                                            </configuration>
                                            <goals>
                                                    <goal>one-jar</goal>
                                            </goals>
                                    </execution>
                                    <execution>
                                            <id>build-second</id>
                                            <configuration>
                                                    <mainClass>org.transitime.applications.UpdateTravelTimes</mainClass>
                                                    <attachToBuild>true</attachToBuild>
                                                    <classifier>onejar</classifier>
                                                    <filename>updateTravelTimes.jar</filename>
                                            </configuration>
                                            <goals>
                                                    <goal>one-jar</goal>
                                            </goals>
                                    </execution>
                                    <execution>
                                            <id>build-third</id>
                                            <configuration>
                                                    <mainClass>org.transitime.applications.SchemaGenerator</mainClass>
                                                    <attachToBuild>true</attachToBuild>
                                                    <classifier>onejar</classifier>
                                                    <filename>generateDatabaseSchema.jar</filename>
                                            </configuration>
                                            <goals>
                                                    <goal>one-jar</goal>
                                            </goals>
                                    </execution>
                                    <execution>
                                            <id>build-fourth</id>
                                            <configuration>
                                                    <mainClass>org.transitime.applications.ScheduleGenerator</mainClass>
                                                    <attachToBuild>true</attachToBuild>
                                                    <classifier>onejar</classifier>
                                                    <filename>generateSchedule.jar</filename>
                                            </configuration>
                                            <goals>
                                                    <goal>one-jar</goal>
                                            </goals>
                                    </execution>
                                    <execution>
                                            <id>build-fifth</id>
                                            <configuration>
                                                    <mainClass>org.transitime.applications.GtfsFileProcessor</mainClass>
                                                    <attachToBuild>true</attachToBuild>
                                                    <classifier>onejar</classifier>
                                                    <filename>processGTFSFile.jar</filename>
                                            </configuration>
                                            <goals>
                                                    <goal>one-jar</goal>
                                            </goals>
                                    </execution>
                                    <execution>
                                            <id>build-sixth</id>
                                            <configuration>
                                                    <mainClass>org.transitime.applications.RmiQuery</mainClass>
                                                    <attachToBuild>true</attachToBuild>
                                                    <classifier>onejar</classifier>
                                                    <filename>RmiQeury.jar</filename>
                                            </configuration>
                                            <goals>
                                                    <goal>one-jar</goal>
                                            </goals>
                                    </execution>
                                    <execution>
                                            <id>build-seventh</id>
                                            <configuration>
                                                    <mainClass>org.transitime.applications.CreateAPIKey</mainClass>
                                                    <attachToBuild>true</attachToBuild>
                                                    <classifier>onejar</classifier>
                                                    <filename>createAPIKey.jar</filename>
                                            </configuration>
                                            <goals>
                                                    <goal>one-jar</goal>
                                            </goals>
                                    </execution>
                            </executions>
                    </plugin>
            </plugins>

    </build>
    <profiles>
            <profile>
                    <id>runCore</id>
                    <build>
                            <plugins>
                                    <plugin>
                                            <groupId>org.codehaus.mojo</groupId>
                                            <artifactId>exec-maven-plugin</artifactId>
                                            <version>1.1.1</version>
                                            <executions>
                                                    <execution>
                                                            <phase>test</phase>
                                                            <goals>
                                                                    <goal>java</goal>
                                                            </goals>
                                                            <configuration>
                                                                    <mainClass>org.transitime.applications.Core</mainClass>
                                                                    <arguments>
                                                                            <argument>arg0</argument>
                                                                            <argument>arg1</argument>
                                                                    </arguments>
                                                            </configuration>
                                                    </execution>
                                            </executions>
                                    </plugin>
                            </plugins>
                    </build>
            </profile>
            <profile>
                    <id>dbTest</id>
                    <build>
                            <plugins>
                                    <plugin>
                                            <groupId>org.codehaus.mojo</groupId>
                                            <artifactId>exec-maven-plugin</artifactId>
                                            <version>1.1.1</version>
                                            <executions>
                                                    <execution>
                                                            <phase>test</phase>
                                                            <goals>
                                                                    <goal>java</goal>
                                                            </goals>
                                                            <configuration>
                                                                    <mainClass>org.transitime.applications.DbTest</mainClass>
                                                                    <arguments>
                                                                            <argument>arg0</argument>
                                                                            <argument>arg1</argument>
                                                                    </arguments>
                                                            </configuration>
                                                    </execution>
                                            </executions>
                                    </plugin>
                            </plugins>
                    </build>
            </profile>
    </profiles>
```

</project>

**Comment by kevin-lee** [comment]
@skibu I needed it for a Scala project and that's it. :smile: OK, that's good. As far as I know, `onejar-maven-plugin` has an option to specify the version of `one-jar-boot`. 

Can you add the following line inside `<configuration>`?

``` xml
<onejarVersion>0.97.3</onejarVersion>
```

That's the way to specify the version of `one-jar-boot`. I hope this solves your issue.

**Comment by skibu** [comment]
I tried adding

<onejarVersion>0.97.3</onejarVersion>

but got the error:

[ERROR] Failed to execute goal
org.dstovall:onejar-maven-plugin:1.4.4:one-jar (build-first) on project
transitimeCore: Execution build-first of goal
org.dstovall:onejar-maven-plugin:1.4.4:one-jar failed: in is null -> [Help
1]

The maven output shows that it is trying to One-Jar version 0.97.3
    [INFO] Using One-Jar version: 0.97.3

In more detail, I changed the pom.xml file to have:
 <execution>
     <id>build-first</id>
     <configuration>
         
         <onejarVersion>0.97.3</onejarVersion>
         <mainClass>org.transitime.applications.Core</mainClass>
         <attachToBuild>true</attachToBuild>
         <classifier>onejar</classifier>
         <filename>core.jar</filename>
     </configuration>
     <goals>
         <goal>one-jar</goal>
     </goals>
  </execution>

I'm thinking the system is now trying to use onejar version 0.97.3, which
is the version of the desired one-jar-boot. But the version of onejar
itself is 0.97.

Any suggestions?
